### PR TITLE
Feature api changes needed #201

### PR DESCRIPTION
**Pull request by lkolacz**
##### Feature list
* Make sure Indicator list API (same used in IP reporting) supports filtering by "reportable id" (anything this indicator has a generic relation with). Eg. a cluster objective with ID XYZ can see indicators associated with it easily. https://projects.invisionapp.com/share/EDBIM0Y8M#/screens/232533791

* Make sure cluster activity API endpoint can be filtered by "cluster objective ID". Since each cluster activity belongs to one specific CO/Cluster objective this should be easy/doable: https://projects.invisionapp.com/share/EDBIM0Y8M#/screens/232533790

* Be able to get a list of partners with appropriate fields in the API GET request for a cluster activity. Basically these are partners who are doing activities whose parent is this cluster activity. Eg. endpoint will be /api/..../cluster-activity/34/partners/ -> it'll return partners that are doing activities whose parent/associated cluster activity is ID 34. UI that needs it is here: https://projects.invisionapp.com/share/EDBIM0Y8M#/screens/232533941

* Be able to filter the partner activity list by partner project id (since each partner activity belongs to one partner project): https://projects.invisionapp.com/share/EDBIM0Y8M#/screens/233065469

* The newly added indicator-reports API endpoint should support "limit" so we can show latest 5-10 on the dashboard like here: https://projects.invisionapp.com/share/EDBIM0Y8M#/screens/232073629

##### Progress checker
- [x] Django

### CODE REVIEW

File: django_api/django_api/apps/partner/views.py [hunk 1]
```diff
@@ -140,3 +142,29 @@ class PartnerSimpleListAPIView(ListAPIView):
...
+class PartnerActivityListCreateAPIView(ListCreateAPIView):
+
+    serializer_class = PartnerActivitySerializer
+    permission_classes = (IsAuthenticated, )
+    pagination_class = SmallPagination
+    filter_backends = (django_filters.rest_framework.DjangoFilterBackend, )
+    filter_class = PartnerActivityFilter
+
+    def get_queryset(self, *args, **kwargs):
+        return PartnerActivity.objects.all()
```
Comment (lkolacz) [inline django_api/django_api/apps/partner/views.py]: queryset will be filtered by Filters called `project` and/or `partner`

**Comment by lkolacz** [comment]
@n1207n Please, take a look so I could merge it tomorrow, thx :)

File: django_api/django_api/apps/cluster/views.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 from rest_framework.views import APIView
 from rest_framework.response import Response
-from rest_framework.generics import ListCreateAPIView
+from rest_framework.generics import ListCreateAPIView, ListAPIView
```
Comment (lkolacz) [inline django_api/django_api/apps/cluster/views.py]: old import.. removed ;-)

File: django_api/django_api/apps/cluster/serializers.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 from rest_framework import serializers
 from core.common import FREQUENCY_LEVEL
+from partner.models import PartnerProject
```
Comment (lkolacz) [inline django_api/django_api/apps/cluster/serializers.py]: old import.. removed ;-)

File: django_api/django_api/apps/core/factories.py [hunk 4]
```diff
@@ -96,6 +96,7 @@ class Meta:
 
 class PartnerActivityFactory(factory.django.DjangoModelFactory):
     title = factory.Sequence(lambda n: "partner_activity_%d" % n)
+    project = factory.SubFactory('core.factories.PartnerProjectFactory')
```
Comment (lkolacz) [inline django_api/django_api/apps/core/factories.py]: linking on fake data generator logic when `Cluster Indicator` is created

**Comment by lkolacz** [comment]
@n1207n Please, take a look again :)

**Comment by n1207n** [comment]
At this moment, I'm sensing this may be duplicate or be overridden to #247. Closing this issue now. We can open a new issue to re-align any missing feature from #247.

**Pull request closed without merging**